x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -384,16 +420,17 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
       },
     };
     const query = getFilteredQuery(dateRangeStart, dateRangeEnd, filters);
-    if (get(query, 'bool.must', undefined)) {
-      query.bool.must.push(statusDown);
+    if (get(query, 'bool.filter', undefined)) {
+      query.bool.filter.push(statusDown);
     } else {
-      set(query, 'bool.must', [{ ...statusDown }]);
+      set(query, 'bool.filter', [{ ...statusDown }]);
     }
 
     const params = {
       index: INDEX_NAMES.HEARTBEAT,
       body: {
         query,
+        size: 0,
         aggs: {
           error_type: {
             terms: {
```

Comment:
This will only give the top 10 `error.type`s, and the top 10 `monitor.id` (and their latest hit).

Reply:
It's not perfect, but it's probably good enough for now. :)